packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -580,5 +578,9 @@ export function getValidatorApi({chain, config, logger, metrics, network, sync}:
 
       network.prepareSyncCommitteeSubnets(subs);
     },
+
+    async prepareBeaconProposer(proposers) {
+      await chain.executionEngine.updateProposerPreparation(chain.clock.currentEpoch, proposers);
```

Comment:
Responding to https://github.com/ChainSafe/lodestar/pull/3958/files#r869010716

Yeah just add this data to the BeaconChain class, it doesn't really concern the executionEngine until block production